app/main.py

```diff
@@ -1,13 +1,42 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": item["line_number"],
+                "column": item["column_number"],
+                "message": item["text"],
+                "name": item["code"],
+                "source": "flake8"
+            } for item in errors
+        ],
+        "path": file_path,
+        "status": "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                {
+                    "line": error["line_number"],
+                    "column": error["column_number"],
+                    "message": error["text"],
+                    "name": error["code"],
+                    "source": "flake8"
+                } for error in errors
+            ],
+            "path": path,
+            "status": "failed" if errors else "passed"
+        } for path, errors in linter_report.items()
+    ]
```

Comment:
In the `format_linter_report` function, you are repeating the same block of code for formatting linter errors. To avoid this repetition, you can use the `format_linter_error` function that you have defined at the top of your file. This way, you can keep your code DRY (Don't Repeat Yourself) and make it more maintainable and easier to read.